src/node/ext/ldap/_node.py

```diff
@@ -207,8 +212,7 @@ def __init__(self, name=None, props=None):
     @finalize
     def __getitem__(self, key):
         # nodes are created for keys, if they do not already exist in memory
-        if isinstance(key, str):
-            key = decode(key)
+        key = decodes(key) if self.root._use_unicode else key
```

Comment:
This kind of code is repeating and blows up several lines.

In order to follow a DRY paradigm, I  propose to pass the current node to ``decodes`` and ``encodes`` and handle the if in there.